src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java

```diff
@@ -74,12 +83,46 @@ public ProfileEntitlementResponse getProfileEntitlements(String bearerToken, Int
     }
 
     @Override
-    public MasterIndexFieldByRecordClassSchemaResponse getMasterIndexFieldByRecordClassSchema(String bearerToken,
-                                                                                              Integer executionId,
-                                                                                              Integer rpoStageId, UserAccountEntity userAccount) {
-        throw new NotImplementedException("Method not implemented");
+    public List<MasterIndexFieldByRecordClassSchema> getMasterIndexFieldByRecordClassSchema(String bearerToken,
+                                                                                            Integer executionId,
+                                                                                            ArmRpoStateEntity rpoStageId,
+                                                                                            UserAccountEntity userAccount) {
+        var armRpoExecutionDetailEntity = armRpoService.getArmRpoExecutionDetailEntity(executionId);
+        armRpoService.updateArmRpoStateAndStatus(armRpoExecutionDetailEntity, rpoStageId,
+                                                 ArmRpoHelper.inProgressRpoStatus(), userAccount);
+
+        MasterIndexFieldByRecordClassSchemaResponse masterIndexFieldByRecordClassSchemaResponse;
+        try {
+            masterIndexFieldByRecordClassSchemaResponse = armRpoClient.getMasterIndexFieldByRecordClassSchema(bearerToken, null);
```

Comment:
Is `null` correct? I think this should be the populated `MasterIndexFieldByRecordClassSchemaRequest`?